src/charm.py

```diff
@@ -86,46 +86,31 @@ def __init__(self, *args):
 
     def reconcile(self, _event: ops.ConfigChangedEvent):
         """Reconcile the entire state of the charm."""
-        statuses: List[StatusBase] = []
+        status_manager = StatusManager()
 
-        try:
+        # Set a default value for any returns in the below context in case of an error
+        prometheus_url = None
+        with status_manager:
             prometheus_url = self._get_prometheus_source_url()
-        except ConfigurationBlockingError as e:
-            statuses.append(ops.BlockedStatus(str(e)))
-            prometheus_url = None
-        except ConfigurationWaitingError as e:
-            statuses.append(ops.WaitingStatus(str(e)))
-            prometheus_url = None
-
-        # prometheus and grafana config are required, but tempo is optional
-        try:
+
+        kiali_config = None
+        with status_manager:
             kiali_config = self._generate_kiali_config(prometheus_url=prometheus_url)
-        except ConfigurationBlockingError as e:
-            statuses.append(ops.BlockedStatus(str(e)))
-            kiali_config = None
 
-        try:
+        with status_manager:
             self._configure_kiali_workload(kiali_config)
-        except ConfigurationWaitingError as e:
-            statuses.append(ops.WaitingStatus(str(e)))
-        except ConfigurationBlockingError as e:
-            statuses.append(ops.BlockedStatus(str(e)))
-
-        if not _is_kiali_available(self._internal_url + self._prefix):
-            statuses.append(
-                ops.WaitingStatus(
-                    "Kiali workload is not available.  See other logs/statuses for reasons why.  If no other statuses"
-                    " are available, this may be transient."
-                )
-            )
 
-        if len(statuses) == 0:
-            statuses.append(ops.ActiveStatus())
+        with status_manager:
+            _is_kiali_available(self._internal_url + self._prefix)
+
+        if len(status_manager) == 0:
+            # Nothing broke, so we can be active
+            status_manager.statuses.append(ops.ActiveStatus())
```

Comment:
Convince my why this should not be done by status manager itself.

Reply:
And if it is, remember to change statuses to _statuses in the lib.

Reply:
That was a mistake on my part - this isn't necessary anymore.  removed